x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/create_datasource_page/step_select_config.tsx

```diff
@@ -134,7 +135,12 @@ export const StepSelectConfig: React.FunctionComponent<{
             </EuiTitle>
           </EuiFlexItem>
           <EuiFlexItem grow={false}>
-            <EuiButtonEmpty iconType="plusInCircle" href={CREATE_NEW_CONFIG_URI} size="s">
+            <EuiButtonEmpty
+              isDisabled={!core.application.capabilities.ingestManager.write}
```

Comment:
Personal opinion: I found this verbose, should we have a hook that hide some of this complexity.
Maybe something like `const hasWriteCapabilites =  useWriteCapabilites()`?

Reply:
I agree it's wordy, but would like to hold off generalizing it until we have a better idea of our permissions model and workflows

Reply:
What I wrote made sense to me at the time, but I now agree we should just pack this up into a hook. I'll send a commit soon.